From a6e82463b04d31f48febfe2455da7f5b0fbba82b Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Fri, 22 Jul 2016 09:07:39 +0200
Subject: [PATCH 3/5] Require pthreads and add optional USE_PTHREAD define

The use of C++11 std::thread requires the -pthread GCC compiler
option. This patch makes pthreads a requirement on non-windows
platforms. Also it checks the CMake variable "USE_PTHREAD" and
- when found - adds a #define it.

Signed-off-by: warped-rudi <r.ihle@s-t.de>
---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8cfda5..71c0259 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,11 @@ if(WIN32)
   endforeach(SHADER_FILE)
   add_custom_target(generate ALL DEPENDS ${SHADER_INCLUDES})
 else()
+  find_package(Threads REQUIRED)
+  add_compile_options(-pthread)
+  if(USE_PTHREAD)
+    add_definitions(-DUSE_PTHREAD)
+  endif()
   find_package(OpenGL)
   if(OPENGL_FOUND)
     add_definitions(-DHAS_OPENGL)
-- 
1.9.1

